File: src/main/java/com/github/njuro/jard/ban/BanResolver.java
package com.github.njuro.jard.ban;

import com.github.njuro.jard.ban.dto.BanDto;
import com.github.njuro.jard.common.Mappings;
import com.github.njuro.jard.utils.PathVariableArgumentResolver;
import java.util.UUID;
import lombok.RequiredArgsConstructor;
import org.jetbrains.annotations.NotNull;
import org.springframework.core.MethodParameter;
import org.springframework.stereotype.Component;
import org.springframework.web.bind.support.WebDataBinderFactory;
import org.springframework.web.context.request.NativeWebRequest;
import org.springframework.web.method.support.ModelAndViewContainer;

/** Resolver for mapping ban id (path variable) to respective {@link BanDto} object */
@Component
@RequiredArgsConstructor
public class BanResolver implements PathVariableArgumentResolver {

  private final BanFacade banFacade;

  @Override
  public boolean supportsParameter(MethodParameter parameter) {
    return parameter.getParameterType().equals(BanDto.class);
  }

  @Override
  public Object resolveArgument(
      @NotNull MethodParameter parameter,
      ModelAndViewContainer mavContainer,
      @NotNull NativeWebRequest webRequest,
      WebDataBinderFactory binderFactory) {
    String id = getPathVariable(Mappings.PLACEHOLDER_BAN, webRequest);

    return banFacade.resolveBan(UUID.fromString(id));
  }
}

File: src/main/kotlin/com/github/njuro/jard/rewrite/ban/BanResolver.kt
package com.github.njuro.jard.rewrite.ban

import com.github.njuro.jard.rewrite.ban.dto.BanDto
import com.github.njuro.jard.rewrite.common.PLACEHOLDER_BAN
import com.github.njuro.jard.rewrite.utils.PathVariableArgumentResolver
import org.springframework.core.MethodParameter
import org.springframework.stereotype.Component
import org.springframework.web.bind.support.WebDataBinderFactory
import org.springframework.web.context.request.NativeWebRequest
import org.springframework.web.method.support.ModelAndViewContainer
import java.util.UUID

/** Resolver for mapping ban id (path variable) to respective [BanDto] object  */
@Component
class BanResolver(private val banFacade: BanFacade) : PathVariableArgumentResolver {
    override fun supportsParameter(parameter: MethodParameter): Boolean =
        parameter.parameterType == BanDto::class.java


    override fun resolveArgument(
        parameter: MethodParameter,
        mavContainer: ModelAndViewContainer?,
        webRequest: NativeWebRequest,
        binderFactory: WebDataBinderFactory?
    ): Any? {
        val id = getPathVariable(PLACEHOLDER_BAN, webRequest)
        return banFacade.resolveBan(UUID.fromString(id))
    }
}
